pyproject.toml

```diff
@@ -236,6 +227,63 @@ filterwarnings = [
 # need to ensure build directories aren't excluded from recursion
 norecursedirs = []
 
+[tool.ruff.lint]
+# E, W   pycodestyle
+# F      Pyflakes
+# UP     pyupgrade
+# B      flake8-bugbear
+# SIM    flake8-simplify
+# YTT    flake8-2020
+# ASYNC  flake8-async
+# C4     flake8-comprehensions
+# I      isort
+extend-select = ["E" ,"W", "F", "UP", "B", "SIM", "YTT", "ASYNC", "C4", "I"]
+# E501      line length
+# SIM905    use list literal instead of str.split
+# C408      use dict literal instead of dict()
+# SIM117    combine with statements
+# B904      raise ... from ...
+# SIM108    use ternary operators
+# C416      unnecessary list comprehension
+# B018      useless expression
+# B019      memory leaks from functools.lru_cache / functools.cache
+# SIM300    Yoda condition (0 != x)
+# C400      unnecessary generator
+# SIM105    use contextlib.suppress() instead of try-except-pass
+# B007      loop control variable not used
+# SIM102    combine if statements
+# SIM115    use context manager for opening files
+# B027      empty method in abstract base class
+# B024      abstract base class without abstract methods / properties
+# SIM103    return condition directly
+# C414      unnecessary list() use within sorted()
+# SIM201    use "x != y" instead of "not x == y"
+# SIM118    use "key in dict" instead of "key in dict.keys()"
```

Comment:
It might be worth commenting on why this list of exceptions. If it's a candidate list for fixes that could be made in the future, it would be good to be explicit about them.